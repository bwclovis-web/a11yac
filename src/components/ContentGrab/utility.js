export const copyToClipboard = (content) => {

    const listener = (evt) => {
        evt.clipboardData.setData("text/html", snaitizeText(content));
        evt.clipboardData.setData("text/plain", snaitizeText(content));
        evt.preventDefault();
    }
    document.addEventListener("copy", listener);
    document.execCommand("copy");
    document.removeEventListener("copy", listener);
}

export const snaitizeText = (text) => {
    var regex = /<br\s*[\/]?>/gi; /<b\s*[\/]?>/gi;
    return text.replace(regex, '   ');
}