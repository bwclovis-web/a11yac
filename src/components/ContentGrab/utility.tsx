interface ClipboardEvent {
    clipboardData: any,
    preventDefault(): void
}

export const copyToClipboard = (content: string | null): void => {
    const listener = (evt: ClipboardEvent) => {
        evt.clipboardData.setData("text/html", content);
        evt.clipboardData.setData("text/plain", content);
        evt.preventDefault();
    }
    document.addEventListener("copy", listener);
    document.execCommand("copy");
    document.removeEventListener("copy", listener);
}
