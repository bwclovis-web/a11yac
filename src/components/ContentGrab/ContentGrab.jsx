import React, {useRef} from 'react';
import styled from 'styled-components';
import UseGetContent from '../../Utils/UseGetContent' 
import {copyToClipboard} from './utility';

const InputStyles = styled.div`
    border: 1px solid var(--grey);
    padding: 1rem 1.4rem;
    position: relative;
    max-height: 230px;
    overflow-Y: auto;

    &._active {
        animation-name: textSelect;
        animation-duration: 1s;
    }

    .alert {
        position: absolute;
    }

    @keyframes textSelect {
        0% {opacity: 1;}
        50% {opacity: 0.3;}
        100% {opacity: 1}
    }

`;

const ContentGrab = ({content}) => {
    const contentRef = useRef(null);
    const {clicked, updateContent} = UseGetContent();
    const handleCopyClick = () => {
        copyToClipboard(contentRef.current.innerHTML);
        updateContent();
    }

    return (
        <div>
            <div className="container">
                <InputStyles ref={contentRef} className={clicked && '_active'} dangerouslySetInnerHTML={{__html: content}} />
                <button type="button" aria-busy={clicked} className="btn" onClick={handleCopyClick}>Copy Acceptance Criteria</button>
            </div>
        </div>
    )
}

export default ContentGrab;