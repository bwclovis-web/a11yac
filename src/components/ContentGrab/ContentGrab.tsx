import React, {useRef, useContext} from 'react';
import styled from 'styled-components';
import {copyToClipboard} from './utility';

import PageContext from '../PageProvider/PageProvder'

interface ConetentI {
    content: string
}

const InputStyles = styled.div<{clicked: boolean}>`
    border: 1px solid var(--grey);
    padding: 1rem 1.4rem;
    position: relative;
    max-height: 230px;
    overflow-Y: auto;
    animation-name: ${props => props.clicked ? 'textSelect' : null};
    animation-duration: 1s;

    @keyframes textSelect {
        0% {opacity: 1;}
        50% {opacity: 0.3;}
        100% {opacity: 1}
    }
`;

const ContentGrab = (props: ConetentI ) => {
    const contentRef = useRef<HTMLDivElement | null>(null);
    const [clicked, updateContent] = useContext(PageContext);

    const handleCopyClick = () => {
        contentRef.current && copyToClipboard(contentRef.current.innerHTML);
        updateContent();
    }

    return (
        <div>
            <div className="container">
                <InputStyles ref={contentRef} clicked={clicked} dangerouslySetInnerHTML={{__html: props.content}} />
                <button type="button" aria-busy={clicked} className="btn" onClick={handleCopyClick}>Copy Acceptance Criteria</button>
            </div>
        </div>
    )
}

export default ContentGrab;