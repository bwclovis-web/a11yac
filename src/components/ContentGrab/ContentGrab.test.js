import React from 'react';
import renderer, {act} from 'react-test-renderer';
import ContentGrab from './ContentGrab';

describe("ContentGrab", () => {
    it('should render correctly ', () => {
        const tree = renderer.create(<ContentGrab/>).toJSON()
        expect(tree).toMatchSnapshot()
    });

    it('should handle click', () => {
        let component;

        act(() => {
            component = renderer.create(<ContentGrab/>)
        })
        expect(component.toJSON()).toMatchSnapshot();

        act(() => {
            component.root.findByType('button').props.onClick();
        })
        expect(component.toJSON()).toMatchSnapshot();
    })
})