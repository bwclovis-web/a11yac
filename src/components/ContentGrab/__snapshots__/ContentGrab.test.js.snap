// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentGrab should handle click 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="sc-bdfBwQ hcjJxQ"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      tabIndex="0"
    />
    <button
      aria-busy={false}
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Copy Acceptance Criteria
    </button>
  </div>
</div>
`;

exports[`ContentGrab should handle click 2`] = `
<div>
  <div
    className="container"
  >
    <div
      className="sc-bdfBwQ hcjJxQ"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      tabIndex="0"
    />
    <button
      aria-busy={false}
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Copy Acceptance Criteria
    </button>
  </div>
</div>
`;

exports[`ContentGrab should render correctly  1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="sc-bdfBwQ hcjJxQ"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
      tabIndex="0"
    />
    <button
      aria-busy={false}
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Copy Acceptance Criteria
    </button>
  </div>
</div>
`;
